src/OpenTelemetry.Contrib.Extensions.AWSXRay/Instrumentation/AWSClientsInstrumentation.cs

```diff
@@ -0,0 +1,33 @@
+﻿// <copyright file="AWSClientsInstrumentation.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using Amazon.Runtime.Internal;
+
+namespace OpenTelemetry.Contrib.Extensions.AWSXRay.Instrumentation
```

Comment:
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/master/src Lets use the naming convention already used in this repo for instrumentations.
OpenTelemetry.Contrib.Instrumentation.AWSXRay ?

Reply:
Good call. And since this instrumentation of AWS SDK clients is not vendor specific to XRay, it makes more sense for the namespace to be OpenTelemetry.Contrib.Instrumentation.AWS

Reply:
You should update the folder name as well.

Reply:
So this might be a bit tricky. My idea is to bundle the AWS related extensions (Id generator, propagator, and instrumentation) in a single package which a user can pull down and use. The AWS instrumentation uses the AWSXRayPropagator. Separating them into different folders (one for extension and one for instrumentation) would require a user to download both the packages. That's why I had the instrumentation as part of the extension, although it did not follow the common structure of other instrumentations.

Reply:
@cijothomas @eddynaka Any thoughts on this?

Reply:
Possibly we can have the folder structure like this?
```
OpenTelemetry.Contrib.Extensions.AWS (folder)
+---- Trace (folder)
|           +---- AWSXRayIdGenerator.cs
|           +---- AWSXRayPropagator.cs
|           +---- AWSXRayEventSource.cs         
+---- AWS-SDK (folder)
|            +---- All the AWS SDK instrumentation files
+---- Rest of the files like TracerProviderBuilderExtensions.cs, README, .csproj, etc.
```